generated/niscope/niscope.proto

```diff
@@ -999,10 +1089,19 @@ message ConfigureChanCharacteristicsResponse {
 
 message ConfigureClockRequest {
   nidevice_grpc.Session vi = 1;
-  string input_clock_source = 2;
-  string output_clock_source = 3;
-  string clock_sync_pulse_source = 4;
-  bool master_enabled = 5;
+  oneof input_clock_source_enum {
```

Comment:
This is a more significant breaking change than I think we've made in the past. Because the types of the slots have changed in an incompatible way (not just a rename or a type change). So, for example, slot 2 was a string and is now an enum. Slot 5 was a bool and is now a string. If only the name changes (or type changes to a compatible type, like int -> enum), then pre-existing clients still work. But source updates will be required if they re-protoc. But with this type of change, pre-existing clients will not work (random behavior, crash, etc).

Have you talked with @harsha-bhushan about whether we're OK with that? For sure, we would need to highlight it more in our release notes.

I think it's possible to avoid this, but it might be a bit tricky in the codegen. You could add a flag to enum fields for a `compatibility_mode`, and then give them indexes at the end:

```proto
message ConfigureClockRequest {
  nidevice_grpc.Session vi = 1;
  oneof input_clock_source_enum {
    ClockingTerminalValues input_clock_source_mapped = 6;
    string input_clock_source_raw = 2;
  }
  oneof output_clock_source_enum {
    ClockingTerminalValues output_clock_source_mapped = 7;
    string output_clock_source_raw = 3;
  }
  oneof clock_sync_pulse_source_enum {
    ClockingTerminalValues clock_sync_pulse_source_mapped = 8;
    string clock_sync_pulse_source_raw = 4;
  }
  bool master_enabled = 5;
}
```

at least .... I think you can do that.

Reply:
There's a little more detail on `oneof` compatibility here: https://developers.google.com/protocol-buffers/docs/proto3#backwards-compatibility_issues . That would only apply if you took my alternative above of keeping the numbering. I think adding a field to a `oneof` is safe, but you should test the case where an old client didn't set the field. That shouldn't cause insane behavior, but it might change behavior (from "default value" to "one of case -> none").

Reply:
My recommendation - make it as less breaking as possible.
Between codegen & backward compatibility for existing clients, we should prefer the latter. Client source breaking on re-protoc is still OK to an extent but unexpected behavior/ breaking compatibility without re-protoc is not OK.

Reply:
I was able to get the correct less breaking behavior of maintaining the field numbers using the 'grpc_raw_field_number" and "grpc_mapped_field_number"

Reply:
Looks great! That fix is much nicer than what I had in mind.

Did you test with an old client and confirm the behavior was OK? Can you do that and update the `Testing Done` to describe what you did?

Reply:
I was able to get this testing done though it took a little more time than expected. I put a new version of the server (containing these changes) on an lvtest machine and connected to it and tested running a simple init configure clock with the new builds on my dev machine client that all worked as expected. On my dev machine I then reverted back to main and rebuilt everything. I ran locally with old client and server to verify just the string passing behavior and that passed as expected. Then I used the old client running simple init/configure clock to connect to the new server on the lvtest machine and that passed successfully so I think we have avoided the massively breaking changes!